src/api/app/components/bs_request_description_component.html.haml

```diff
@@ -0,0 +1,36 @@
+- types.each_with_index do |(type, actions), index|
+  - case type
+  - when 'submit'
+    Submit
+    = sanitize(source_and_target_container(actions))
+  - when 'delete'
+    Delete
+    = sanitize(delete_target(actions))
```

Comment:
This very much sounds like `BsRequestActionSubmit` and `BsRequestActionDelete` and ... code to me.